cmd/rook/ctl/multus/validation/validation.go

```diff
@@ -0,0 +1,216 @@
+	cleanupCmd = &cobra.Command{
+		Use:   "cleanup",
+		Short: "Clean up Multus validation test resources",
+		Run: func(cmd *cobra.Command, args []string) {
+			runCleanup(cmd.Context())
+		},
+		Args: cobra.NoArgs,
+	}
+)
+
+func init() {
+	validationConfig.Clientset = client.NewClientset()
+
+	Cmd.AddCommand(runCmd)
+	Cmd.AddCommand(cleanupCmd)
+
+	// flags on ALL subcommands - makes output more straightforward than using PersistentFlags() global flags on parent
+	for _, subCommand := range Cmd.Commands() {
+		// the default namespace is the current namespace the operator pod is running in if possible
+		defaultNamespace := os.Getenv(k8sutil.PodNamespaceEnvVar)
+		if defaultNamespace == "" {
+			defaultNamespace = DefaultValidationNamespace
+		}
+		subCommand.Flags().StringVarP(&validationConfig.Namespace, "namespace", "n", defaultNamespace,
+			"The namespace for validation test resources. "+
+				"It is recommended to set this to the namespace in which Rook's Ceph cluster will be installed.")
+
+		// VarPF() keeps the the specific var passed to it for setting at runtime, and the current
+		// val of that var when VarPF() is called is used as the default
+		validationConfig.ResourceTimeout = time.Duration(DefaultTimeoutMinutes) * time.Minute
+		t := (*timeoutMinutes)(&validationConfig.ResourceTimeout)
+		subCommand.Flags().VarPF(t, "timeout-minutes", "", /* no shorthand */
+			"The time to wait for resources to change to the expected state. For example, for the "+
+				"test web server to start, for test clients to become ready, or for test resources to be deleted. "+
+				"At longest, this may need to reflect the time it takes for client pods to to pull images, "+
+				"get address assignments, and then for each client to determine that its network connection is stable. "+
+				"Minimum: 1 minute. Recommended: 2 minutes or more.")
+	}
+
+	// flags for 'validation run'
+	runCmd.Flags().StringVar(&validationConfig.PublicNetwork, "public-network", "",
+		"The name of the Network Attachment Definition (NAD) that will be used for Ceph's public network. "+
+			"This should be a namespaced name in the form <namespace>/<name> if the NAD is defined in a different namespace from the cluster namespace.")
+	runCmd.Flags().StringVar(&validationConfig.ClusterNetwork, "cluster-network", "",
+		"The name of the Network Attachment Definition (NAD) that will be used for Ceph's cluster network. "+
+			"This should be a namespaced name in the form <namespace>/<name> if the NAD is defined in a different namespace from the cluster namespace.")
+	runCmd.Flags().IntVar(&validationConfig.DaemonsPerNode, "daemons-per-node", DefaultDaemonsPerNode,
+		"The number of validation test daemons to run per node. "+
+			"It is recommended to set this to the maximum number of Ceph daemons that can run on any node in the worst case of node failure(s). "+
+			"The default value is set to the worst-case value for a Rook Ceph cluster with 3 portable OSDs, 3 portable monitors, "+
```

Comment:
That assumes a production cluster like used by ODF. Other users might wish to allow portable mon colocation in some scenarios. I would much rather assume the worst possible of the worst, and have users scale it down for themselves if they get false negatives; rather than give too many false positives.

Reply:
Makes sense. And I see the provisioners only colocate on single-node environments, so unlikely to be necessary for users who want multus.